ios/FluentUI/DrawerVnext/Drawer.swift

```diff
@@ -0,0 +1,170 @@
+    @Published public var shadowDepthX: CGFloat!
+    @Published public var shadowDepthY: CGFloat!
+
+    public init() {
+        self.themeAware = true
+        didChangeAppearanceProxy()
+    }
+
+    @objc open func didChangeAppearanceProxy() {
+        let appearanceProxy = StylesheetManager.S.DrawerTokens
+        shadowColor = Color(appearanceProxy.shadowColor)
+        shadowOpacity = Double(appearanceProxy.shadowOpacity)
+        shadowBlur = appearanceProxy.shadowBlur
+        shadowDepthX = appearanceProxy.shadowX
+        shadowDepthY = appearanceProxy.shadowY
+    }
+}
+
+// MARK: - Drawer
+
+/// `Drawer` is used to present a overlay a content partially on another view.
+/// `Drawer`  support horizontal axis and is expanded by default from left side of the screen unless explicitly specified
+///  Set `Content` to provide content for the drawer.
+public struct Drawer<Content: View>: View {
+
+    // content view on top of `Drawer`
+    public var content: Content
+
+    // configure the behavior of drawer
+    @ObservedObject public var state = DrawerState()
+
+    // configure the apperance of drawer
+    @ObservedObject public var tokens = DrawerTokens()
+
+    // keep track of dragged offset
+    @State internal var draggedOffsetWidth: CGFloat?
+
+    // internal drawer state
+    @State internal var isDrawerOpen: Bool = false
+
+    public var body: some View {
+        GeometryReader { proxy in
+            SlideOverPanel(
+                content: content,
+                isOpen: $isDrawerOpen,
+                preferredContentOffset: $draggedOffsetWidth)
+                .backgroundOpactiy(backgroundLayerOpacity)
+                .direction(slideOutDirection)
+                .width(proxy.portraitOrientationAgnosticSize().width)
+                .performOnBackgroundTap {
+                    state.isExpanded.toggle()
+                }
+                .onReceive(state.$isExpanded, perform: { value in
+                    withAnimation {
+                        self.isDrawerOpen = value
+                        self.draggedOffsetWidth = nil // drag ends
+                    }
+                })
+                .gesture(dragGesture(screenWidth: proxy.portraitOrientationAgnosticSize().width))
+        }
+        .edgesIgnoringSafeArea(.all)
+    }
+
+    private var backgroundLayerOpacity: Double {
+        return state.backgroundDimmed ? 0.5 : 0
+    }
+
+    private var slideOutDirection: SlideOverDirection {
+        return state.presentationDirection == .left ? .left : .right
+    }
+}
+
+// MARK: - Utility + Helpers
+
+extension GeometryProxy {
+    func portraitOrientationAgnosticSize() -> CGSize {
+        let isPortraitMode = self.size.width < self.size.height
+        if isPortraitMode {
+            return CGSize(width: self.size.width, height: self.size.height)
+        } else {
+            return CGSize(width: self.size.height, height: self.size.width)
+        }
+    }
+}
+
+// MARK: - Previews
```

Comment:
Are previews something we want to check in? I think they're a good idea if someone wants to see the live previews within Xcode, but they could also test the demo app.. Thoughts?

Reply:
I agree we should question ourselves about it and it's why I created #360.

My first gut reaction is that this should be part of the demo app code to prevent increasing the library size for consumers out of the cocoapods world.

Reply:
I would like to keep previews for testing till we have a solution.. We can move them to appropriate place when we address #360